public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -294,27 +295,71 @@ export class PanelQueryRunner {
    */
   destroy() {
     // Tell anyone listening that we are done
-    if (this.subject) {
-      this.subject.complete();
-    }
-
-    if (this.subscription) {
-      this.subscription.unsubscribe();
-    }
+    this.subject?.complete();
+    this.subscription?.unsubscribe();
+    this.dashboardSubscription?.unsubscribe();
   }
 
   useLastResultFrom(runner: PanelQueryRunner) {
     this.lastResult = runner.getLastResult();
+    this.lastDashboardResult = runner.getLastDashboardResult();
 
     if (this.lastResult) {
       // The subject is a replay subject so anyone subscribing will get this last result
-      this.subject.next(this.lastResult);
+      this.subject.next(this.decorateWithDashboardResults(this.lastResult));
     }
   }
 
   getLastResult(): PanelData | undefined {
     return this.lastResult;
   }
+
+  getLastDashboardResult(): DashboardQueryRunnerResult {
+    return this.lastDashboardResult;
+  }
+
+  refreshDashboardQuerySubscription() {
+    this.initDashboardQuerySubscription();
+  }
+
+  private decorateWithDashboardResults(data: PanelData | undefined): PanelData | undefined {
+    if (!data) {
+      return data;
+    }
+
+    const { alertState, annotations } = this.lastDashboardResult;
+    data.annotations = [new ArrayDataFrame(annotations)];
+    data.alertState = alertState;
+
+    return data;
+  }
+
+  private initDashboardQuerySubscription() {
+    if (this.dashboardSubscription) {
+      this.dashboardSubscription.unsubscribe();
+    }
+
+    if (!this.dataConfigSource?.getDataSupport) {
+      return;
+    }
+
+    const { alertStates, annotations } = this.dataConfigSource.getDataSupport();
+
+    if (alertStates || annotations) {
+      const id = this.dataConfigSource.editSourceId ?? this.dataConfigSource.id;
+      this.dashboardSubscription = getDashboardQueryRunner()
+        .getResult(id)
+        .subscribe((dashboardResult) => {
+          this.lastDashboardResult = dashboardResult;
+          const data = this.lastResult ?? {
+            timeRange: getDefaultTimeRange(),
+            series: [],
+            state: LoadingState.NotStarted,
+          };
+          this.subject.next(this.decorateWithDashboardResults(data));
```

Comment:
since both this annotation query and the normal query now run in "parallel" observables they will both always trigger an update/render.

Reply:
Damn, you're right! Ok, then I need to rethink this whole PR.